TTCN3_Project/CANFilterControl/com/testingtech/playits/canfilter/Car2XControlExternalFunctionProvider.java

```diff
@@ -0,0 +1,97 @@
+package com.testingtech.playits.canfilter;
+
+import java.io.IOException;
+
+import org.etsi.ttcn.tci.BooleanValue;
+import org.etsi.ttcn.tci.CharstringValue;
+import org.etsi.ttcn.tci.IntegerValue;
+
+import com.testingtech.ttcn.annotation.ExternalFunction;
+import com.testingtech.ttcn.tri.AnnotationsExternalFunctionPlugin;
+import com.testingtech.util.UUID;
+
+@ExternalFunction.Definitions(Car2XControlExternalFunctionProvider.class)
+public class Car2XControlExternalFunctionProvider extends
+    AnnotationsExternalFunctionPlugin {
+
+  private static Process process;
+  private static String processId;
+
+  /**
+   * Starts a CAN filter service process with specified host and port
+   * 
+   * @param host
+   *            host name
+   * @param portNumber
+   *            port
+   * @return process id
+   */
+  @ExternalFunction(name = "start_Filter", module = "Car2X_Control")
+  public CharstringValue start_Filter(CharstringValue host,
+      IntegerValue portNumber) {
+    return start_Filter(
+        // CANFilterService.class.getCanonicalName()
+        "com.testingtech.playits.canfilter.CANFilterService",
+        new String[] { host.getString(),
+            String.valueOf(portNumber.getInt()) });
+  }
+
+  private CharstringValue start_Filter(String mainClass, String[] parameters) {
+    if (process != null && process.isAlive()) {
+      return newCharstringValue(processId);
+    }
+    processId = UUID.randomUUID().toString();
+
+    logInfo("Starting " + mainClass + " with parameters: \""
+        + join(parameters, "\" \"") + "\"");
+
+    try {
+      process = Runtime.getRuntime().exec("java -classpath build/CANFilter;libs/java-json.jar "
+              + mainClass + " " + join(parameters, " "));
+
+      new StreamForwarder(process.getInputStream(), System.out)
```

Comment:
I actually don't see how you can close these forwarded streams. Does this happen automatically, when the process is closed? I don't think that it's a good idea to  just create objects via "new Object()" without assigning it to a specific variable, since you lose every control then.